shared/actions/fs/index.js

```diff
@@ -315,6 +303,31 @@ function* ignoreFavoriteSaga(action: FsGen.FavoriteIgnorePayload): Saga.SagaGene
   }
 }
 
+function* onAction(action: FsGen.OnActionPayload): Saga.SagaGenerator<any, any> {
+  const {path, type, targetRect} = action.payload
+  console.log(action.payload)
+  // We may not have the folder loaded yet, but will need metadata to know
+  // folder entry types in the popup. So dispatch an action now to load it.
+  type === 'folder' && (yield Saga.put(FsGen.createFolderListLoad({path})))
+  yield Saga.put(
+    navigateAppend([
+      {
+        props: {
+          path,
+          position: 'bottom right',
+          isShare: false,
+          targetRect,
+        },
+        selected: 'pathItemAction',
+      },
+    ])
+  )
+}
+
+function* metaSaga(): Saga.SagaGenerator<any, any> {
```

Comment:
What is `metaSaga`?

Reply:
This includes tasks that either triggers other Saga actions, or route changes. I struggled with naming this one. Would you mind suggesting a better name?

Reply:
It looks to me that `metaSaga` isn't actually needed. You can put its `safeTakeEvery`s into the place where `subSaga` is yielded.

Reply:
I was trying to separate these from the others, since they have a dependency on other saga tasks.

Reply:
I think that adds more confusion than it helps.